swarm/pss/pss.go

```diff
@@ -472,13 +465,22 @@ func (p *Pss) process(pssmsg *PssMsg, raw bool, prox bool) error {
 	p.executeHandlers(psstopic, payload, from, raw, prox, asymmetric, keyid)
 
 	return nil
+}
 
+// copy all registered handlers for respective topic in order to avoid data race or deadlock
+func (p *Pss) getHandlers(topic Topic) (ret []*handler) {
+	p.handlersMu.RLock()
+	defer p.handlersMu.RUnlock()
+	for k, _ := range p.handlers[topic] {
```

Comment:
This is getting a bit weird. We first hold the handlers in a map (handler -> bool), and now we transition to a slice of handlers.

We should probably unify those data structures. I think `p.handlers` should also be a slice of handlers.

Out of scope for this PR.

Reply:
the bool is not used in this particular function, but still used elsewhere